File: app/src/main/java/org/jqt3of5/android/dicebag/room/DiceInPlayEntity.java
package org.jqt3of5.android.dicebag.room;

import android.arch.persistence.room.*;

/**
 * Created by Brittany on 4/28/2018.
 */
@Entity(tableName = "DiceInPlay")
public class DiceInPlayEntity
{
    @PrimaryKey(autoGenerate = true)
    public int diceId;

    @ForeignKey(entity= DiceTemplateEntity.class,
            parentColumns = {"templateId"},
            childColumns = {"diceTemplateId"}
            /*, onDelete = ForeignKey.CASCADE*/)
    public int diceTemplateId;

    @ColumnInfo
    public String description;

    @ColumnInfo
    public String value;

    @ColumnInfo
    public int mode;
}

File: app/src/main/java/org/jqt3of5/android/dicebag/DiceInPlayEntity.kt
package org.jqt3of5.android.dicebag

import android.arch.persistence.room.*

/**
 * Created by Brittany on 4/28/2018.
 */
@Entity(tableName = "DiceInPlay")
class DiceInPlayEntity
{
    @PrimaryKey
    public var diceId : Int = 0

    @ForeignKey(entity= DiceTemplateEntity::class,
            parentColumns = arrayOf("diceTemplateId"),
            childColumns = arrayOf("diceTemplateId")
            /*, onDelete = ForeignKey.CASCADE*/)
    public var diceTemplateId : Int = 0

    @ColumnInfo(name = "description")
    public var description : String = ""

    @ColumnInfo(name = "value")
    public var value : String = ""

    @ColumnInfo(name = "mode")
    public var mode : Int = 0

}